wolff/optimization/finite_difference_gradient.py

```diff
@@ -0,0 +1,51 @@
+'''Implementation of a regularized level-set model with laplace operator'''
+from wolff.layer.layer import Layer
+
+
+class FiniteDifferenceGradient(Layer):
+    def __init__(self, J=[], J_dtheta=[], dtheta=[]):
+        super(FiniteDifferenceGradient, self).__init__()
+        self.J = J
+        self.J_dtheta = J_dtheta
+        self.dtheta = dtheta
+
+    @property
+    def dtheta(self):
+        return self._dtheta
+
+    @dtheta.setter
+    def dtheta(self, value):
+        if type(value) is float:
+            if value < 0:
+                raise ValueError('dtheta must be greater than zero')
+        elif any(i < 0 for i in value):
+            raise ValueError('dtheta must be greater than zero')
+        self._dtheta = value
+
+    @property
+    def J(self):
+        return self._J
+
+    @J.setter
+    def J(self, value):
+        self._J = value
```

Comment:
Since this is a list, we need a deep copy.
```suggestion
        self._J = list(value)
```